lib/krane/restart_task.rb

```diff
@@ -42,7 +42,7 @@ def initialize(context:, namespace:, logger: nil, max_watch_seconds: nil)
     def run(*args)
       perform!(*args)
       true
-    rescue FatalDeploymentError
+    rescue Krane::FatalDeploymentError, KubernetesDeploy::FatalDeploymentError
```

Comment:
Why do you need to add the `KuberentesDeploy` versions to these? I would expect that everything is actually raising the `Krane` versions, so things under the Krane namespace can ignore the fact that the KD versions exist.

Reply:
Because of https://github.com/Shopify/kubernetes-deploy/pull/585/commits/9042b22beec335efc66601b6896c8c22ea062d81#diff-d6e404f21ee0321894c447fb753b8423L153, which was:

```ruby
    def run(*args)
      run!(*args)
      true
    rescue FatalDeploymentError
      false
    end
```

The thing is that in this context, `FatalDeploymentError` is actually `Krane::FatalDeploymentError`, and as the exception is changed to `KubernetesDeploy::FatalDeploymentError` due to the recent changes it was not being caught anymore, failing the tests related to this area.

Reply:
Oh, why is the error conversion in the shared superclass? Can't you do it directly in `KubernetesDeploy::DeployTask` and avoid this problem?

Reply:
hmmmm, let me try!

Reply:
Thanks for the feedback, I think it should work now.